### PR TITLE
Fix throwing client exception.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Issue:
- https://github.com/wso2/product-is/issues/19466

Throw and log an client execption if the debug logs enabled.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+package org.wso2.carbon.identity.oauth2;
+
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+
+/**
+ * OAuth client exception extended from OAuthSystemException.
+ */
+public class OAuthOltuClientException extends OAuthSystemException {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java:26]: Lets have a general class name maybe OAuthSystemClientException

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    public OAuthOltuClientException(String message) {
+        super(message);
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java:29]: formatting

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java [hunk 3]
```diff
@@ -238,12 +240,18 @@ private String getBindingRefFromType(HttpServletRequest request, String consumer
...
         TokenBinder tokenBinder = tokenBinderOptional.get();
-        String tokenBindingRef = OAuth2Util.getTokenBindingReference(tokenBinder.getTokenBindingValue(request));
-        if (StringUtils.isBlank(tokenBindingRef)) {
-            throw new IdentityOAuth2Exception("Token binding reference is null for the application " +
-                    consumerKey + " with binding type " + bindingType + ".");
+
+        try {
+            String tokenBindingRef = OAuth2Util.getTokenBindingReference(tokenBinder.getTokenBindingValue(request));
+            if (StringUtils.isBlank(tokenBindingRef)) {
+                throw new IdentityOAuth2Exception("Token binding reference is null for the application " +
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java:247]: shouldn't this also be a client exception?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java]: Addressed.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuthOltuClientException.java]: Addressed.

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**